lib/api/apiUtils/object/versioning.js

```diff
@@ -434,6 +434,57 @@ function preprocessingVersioningDelete(bucketName, bucketMD, objectMD, reqVersio
     return options;
 }
 
+/**
+ * Keep metadatas when the object is restored from cold storage
+ * but remove the specific ones we don't want to keep
+ * @param {object} objMD - obj metadata
+ * @param {object} metadataStoreParams - custom built object containing resource details.
+ * @return {undefined}
+ */
+function restoreMetadatas(objMD, metadataStoreParams) {
+    /* eslint-disable no-param-reassign */
+    const userMDToSkip = ['x-amz-meta-scal-s3-restore-attempt'];
+    // We need to keep user metadata and tags
+    Object.keys(objMD).forEach(key => {
+        if (key.startsWith('x-amz-meta-') && !userMDToSkip.includes(key)) {
+            metadataStoreParams.metaHeaders[key] = objMD[key];
+        } else if (key.startsWith('content-')) {
+            metadataStoreParams[key] = objMD[key];
+        }
+    });
+
+    if (objMD['x-amz-server-side-encryption']) {
```

Comment:
In my opinion, we should not "copy" any metadata that is data-related (data encryption, data size...).